apps/design-system/src/subjects/views/execution/pipeline-execution-graph.tsx

```diff
@@ -60,9 +69,36 @@ export function StepNodeComponent({
 }: {
   node: LeafNodeInternalType<StepNodeDataType>
 } & NodeProps) {
-  const { name, icon } = node.data
+  const { name, icon, logs, stage } = node.data
 
-  return <PipelineNodes.StepNode name={name} icon={icon} onEllipsisClick={() => undefined} readonly={readonly} />
+  return (
+    <Drawer.Root direction="right">
+      <Drawer.Trigger asChild>
+        <PipelineNodes.StepNode name={name} icon={icon} onEllipsisClick={() => undefined} readonly={readonly} />
```

Comment:
as discussed, we can probably replace the `readonly` prop with a `mode: Edit | Execution` kind of a prop

Reply:
Done.